test/unit/org/apache/cassandra/service/reads/DigestResolverTest.java

```diff
@@ -78,6 +80,70 @@ public void noRepairNeeded()
         assertPartitionsEqual(filter(iter(response)), resolver.getData());
     }
 
+    @Test
+    public void multiThreadedNoRepairNeededReadCallback() throws InterruptedException
+    {
+        SinglePartitionReadCommand command = SinglePartitionReadCommand.fullPartitionRead(cfm, nowInSec, dk);
+        EndpointsForToken targetReplicas = EndpointsForToken.of(dk.getToken(), full(EP1), full(EP2));
+        PartitionUpdate response = update(row(1000, 4, 4), row(1000, 5, 5)).build();
+        ReplicaPlan.SharedForTokenRead plan = plan(ConsistencyLevel.ONE, targetReplicas);
+
+        ExecutorService pool = Executors.newFixedThreadPool(4);
```

Comment:
Nit: wouldn't it be enough with 2 threads?

Reply:
Yeah, I think this might just be an artifact from before I added the start latch earlier.